channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/environment/JiraServerEnvironmentVariableHandlerFactory.java

```diff
@@ -0,0 +1,90 @@
+    public static final String DISABLE_PLUGIN_KEY = "ALERT_CHANNEL_JIRA_SERVER_JIRA_SERVER_DISABLE_PLUGIN_CHECK";
+    public static final String PASSWORD_KEY = "ALERT_CHANNEL_JIRA_SERVER_JIRA_SERVER_PASSWORD";
+    public static final String URL_KEY = "ALERT_CHANNEL_JIRA_SERVER_JIRA_SERVER_URL";
+    public static final String USERNAME_KEY = "ALERT_CHANNEL_JIRA_SERVER_JIRA_SERVER_USERNAME";
+
+    public static final Set<String> VARIABLE_NAMES = Set.of(DISABLE_PLUGIN_KEY, PASSWORD_KEY, URL_KEY, USERNAME_KEY);
+
+    private final JiraServerGlobalConfigAccessor configAccessor;
+    private final EnvironmentVariableUtility environmentVariableUtility;
+
+    @Autowired
+    public JiraServerEnvironmentVariableHandlerFactory(JiraServerGlobalConfigAccessor configAccessor, EnvironmentVariableUtility environmentVariableUtility) {
+        this.configAccessor = configAccessor;
+        this.environmentVariableUtility = environmentVariableUtility;
+    }
+
+    @Override
+    public EnvironmentVariableHandler build() {
+        return new EnvironmentVariableHandler(ChannelKeys.JIRA_SERVER.getDisplayName(), VARIABLE_NAMES, this::isConfigurationMissing, this::updateConfiguration);
+    }
+
+    private Boolean isConfigurationMissing() {
+        return configAccessor.getConfigurationCount() <= 0;
+    }
+
+    private Properties updateConfiguration() {
+        Properties properties = new Properties();
+        String url = environmentVariableUtility.getEnvironmentValue(URL_KEY).orElse(null);
+
+        if (StringUtils.isBlank(url)) {
+            return properties;
+        }
+
+        String name = AlertRestConstants.DEFAULT_CONFIGURATION_NAME;
+        String userName = environmentVariableUtility.getEnvironmentValue(USERNAME_KEY).orElse(null);
+        String createdAt = DateUtils.formatDate(DateUtils.createCurrentDateTimestamp(), DateUtils.UTC_DATE_FORMAT_TO_MINUTE);
+        JiraServerGlobalConfigModel configModel = new JiraServerGlobalConfigModel(null, name, createdAt, createdAt, url, userName);
+        environmentVariableUtility.getEnvironmentValue(DISABLE_PLUGIN_KEY)
+            .map(Boolean::valueOf)
+            .ifPresent(configModel::setDisablePluginCheck);
+
+        environmentVariableUtility.getEnvironmentValue(PASSWORD_KEY)
+            .ifPresent(configModel::setPassword);
+
+        JiraServerGlobalConfigModel obfuscatedModel = configModel.obfuscate();
+        properties.put(URL_KEY, obfuscatedModel.getUrl());
+        if (StringUtils.isNotBlank(obfuscatedModel.getUserName())) {
+            properties.put(USERNAME_KEY, obfuscatedModel.getUserName());
+        }
+        obfuscatedModel.getDisablePluginCheck().map(String::valueOf).ifPresent(value -> properties.put(DISABLE_PLUGIN_KEY, value));
+        Boolean passwordSet = obfuscatedModel.getPasswordSet().orElse(Boolean.FALSE);
+
+        if (Boolean.TRUE.equals(passwordSet)) {
+            properties.put(PASSWORD_KEY, AlertRestConstants.MASKED_VALUE);
```

Comment:
This line is a bit concerning to me. Is there a way we could modify this to where we only work with classes that implement obfuscate? That way we can call it in the generic class ensuring we are always working with obfuscated models. My concern is someone forgetting to manually call obfuscate when implementing another channel and accidentally leaking private data in the logs.

Reply:
I would like to change the object returned by the environment handler into something that better represents the data than just properties.  Especially now that this has come up.  That would mean changes to both the email channel and jira channel so I want to do that in a different PR. 
Seemed a bit out of scope for this PR.